# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1424415251 18000
#      Fri Feb 20 01:54:11 2015 -0500
# Node ID 166051951878a4a6fd1be493ffd82cc9332e4546
# Parent  bf737d9277074e027f8d810c40319a2027307e15
avoid warning from GCC about variables possibly clobbered by longjmp

* __glpk__.cc (F__glpk__): Declare errnum volatile to avoid GCC
warning.

diff --git a/libinterp/dldfcn/__glpk__.cc b/libinterp/dldfcn/__glpk__.cc
--- a/libinterp/dldfcn/__glpk__.cc
+++ b/libinterp/dldfcn/__glpk__.cc
@@ -711,17 +711,18 @@ Undocumented internal function.\n\
   OCTAVE_GLPK_GET_REAL_PARAM ("tolobj", par.tolobj);
 
   //-- Assign pointers to the output parameters
   ColumnVector xmin (mrowsc, octave_NA);
   double fmin = octave_NA;
   ColumnVector lambda (mrowsA, octave_NA);
   ColumnVector redcosts (mrowsc, octave_NA);
   double time;
-  int status, errnum = 0;
+  int status;
+  volatile int errnum = 0;
 
   int jmpret = setjmp (mark);
 
   if (jmpret == 0)
     errnum = glpk (sense, mrowsc, mrowsA, c, nz, rn.fortran_vec (),
                    cn.fortran_vec (), a.fortran_vec (), b, ctype,
                    freeLB.fortran_vec (), lb, freeUB.fortran_vec (), ub,
                    vartype.fortran_vec (), isMIP, lpsolver, save_pb, scale,
